lib/gems/pending/util/object_storage/miq_swift_storage.rb

```diff
@@ -0,0 +1,148 @@
+        :path          => "#{Fog::OpenStack.escape(swift_file.directory.key)}/#{Fog::OpenStack.escape(swift_file.key)}"
+      }
+      swift_file.service.send(:request, params)
+      clear_split_vars
+    rescue Excon::Errors::Unauthorized => err
+      logger.error("Access to Swift container #{@container_name} failed due to a bad username or password. #{err}")
+      msg = "Access to Swift container #{@container_name} failed due to a bad username or password. #{err}"
+      raise err, msg, err.backtrace
+    rescue => err
+      logger.error("Error uploading #{source_input} to Swift container #{@container_name}. #{err}")
+      msg = "Error uploading #{source_input} to Swift container #{@container_name}. #{err}"
+      raise err, msg, err.backtrace
+    end
+  end
+
+  def mkdir(_dir)
+    container
+  end
+
+  def container
+    @container ||= begin
+                     container   = swift.directories.get(container_name)
+                     logger.debug("Swift container [#{container}] found") if container
+                     container ||= create_container
+                     container
+                   rescue Fog::Storage::OpenStack::NotFound
+                     logger.debug("Swift container #{container_name} does not exist.  Creating.")
+                     create_container
+                   rescue => err
+                     logger.error("Error getting Swift container #{container_name}. #{err}")
+                     msg = "Error getting Swift container #{container_name}. #{err}"
+                     raise err, msg, err.backtrace
+                   end
+  end
+
+  private
+
+  def swift
+    return @swift if @swift
+    require 'manageiq/providers/openstack/legacy/openstack_handle'
+    extra_options = {}
+    extra_options[:domain_id] = @domain_id
+    extra_options[:service] = "Compute"
+     @osh ||= OpenstackHandle::Handle.new(@username, @password, @host, @port, @api_version, @security_protocol, extra_options)
+    @osh.connection_options = {:instrumentor => $fog_log}
+    begin
+      @swift ||= @osh.swift_service
+    rescue Excon::Errors::Unauthorized => err
+      logger.error("Access to Swift host #{@host} failed due to a bad username or password. #{err}")
+      msg = "Access to Swift host #{@host} failed due to a bad username or password. #{err}"
+      raise err, msg, err.backtrace
+    rescue => err
+      logger.error("Error connecting to Swift host #{@host}. #{err}")
+      msg = "Error connecting to Swift host #{@host}. #{err}"
+      raise err, msg, err.backtrace
+    end
+  end
+
+  def create_container
+    container = swift.directories.create(:key => container_name)
+    logger.debug("Swift container [#{container}] created")
+    container
+  rescue => err
+    logger.error("Error creating Swift container #{container_name}. #{err}")
+    msg = "Error creating Swift container #{container_name}. #{err}"
+    raise err, msg, err.backtrace
+  end
+
+  def download_single(source, destination)
```

Comment:
Maybe raise an error in here for now instead of implementing it with `s3` code 😉